test/e2e/tests/test_repository.py

```diff
@@ -365,3 +366,56 @@ def test_repository_policy(self, ecr_client):
         # Check ECR repository doesn't exists
         exists = self.repository_exists(ecr_client, resource_name)
         assert not exists
+
+    def test_repository_create_will_all_fields(self, ecr_client):
+        resource_name = random_suffix_name("ecr-repository", 24)
+
+        replacements = REPLACEMENT_VALUES.copy()
+        replacements["REPOSITORY_NAME"] = resource_name
+        replacements["REPOSITORY_POLICY"] = REPOSITORY_POLICY_GET_DOWNLOAD_URL_ALL
+        replacements["REPOSITORYLIFECYCLE_POLICY"] = LIFECYCLE_POLICY_FILTERING_ON_IMAGE_AGE
+        replacements["REGISTRY_ID"] = get_account_id()
+
+        # Load Repository CR
+        resource_data = load_ecr_resource(
+            "repository_policy",
+            additional_replacements=replacements,
+        )
+        logging.debug(resource_data)
+
+        # Create k8s resource
+        ref = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL,
+            resource_name, namespace="default",
+        )
+        k8s.create_custom_resource(ref, resource_data)
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        assert cr is not None
+        assert k8s.get_resource_exists(ref)
+
+        time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+        # Get latest repository CR
+        cr = k8s.wait_resource_consumed_by_controller(ref)
```

Comment:
I'm aware that this ideally should be a fixture, but we probably want to do in a separate PR to refactor all the e2e tests in the ecr-controller and other controllers as well..